src/explore-education-statistics-admin/src/components/editable/FeaturedTableLinkInsertForm.tsx

```diff
@@ -0,0 +1,148 @@
+      ? featuredTables.filter(table => {
+          return table.name.toLowerCase().includes(value.toLowerCase());
+        })
+      : [];
+    setSearchResults(nextSearchResults);
+  }, 400);
+
+  const handleFormSubmit = (values: FormValues) =>
+    onSubmit({
+      text: values.text,
+      url: `${config.PublicAppUrl}/data-tables/fast-track/${values.dataBlockParentId}?featuredTable=true`,
+    });
+
+  const validationSchema = useMemo<ObjectSchema<FormValues>>(() => {
+    return Yup.object({
+      dataBlockParentId: Yup.string().required('Select a featured table'),
+      text: Yup.string().required('Enter link text'),
+    });
+  }, []);
+
+  return (
+    <FormProvider validationSchema={validationSchema}>
+      {({ formState, getValues, handleSubmit, setValue, trigger }) => {
+        const { getError } = createRHFErrorHelper({
+          errors: formState.errors,
+          touchedFields: formState.touchedFields,
+          isSubmitted: true,
+        });
+
+        return (
+          <RHFForm
+            id="featuredTablesForm"
+            showErrorSummary={false}
+            onSubmit={handleFormSubmit}
+          >
+            {featuredTables.length > 0 ? (
+              <>
+                <FormComboBox
+                  error={getError('dataBlockParentId')}
+                  hideSearchIcon={!!selectedEntry}
+                  id="featuredTablesSearch"
+                  inputLabel="Featured table"
+                  inputValue={selectedEntry?.name}
+                  options={searchResults.map(result => result.name)}
+                  overflowContainer={false}
+                  onInputChange={event => {
+                    search(event.target.value);
+                  }}
+                  onSelect={selectedIndex => {
+                    setValue(
+                      'text' as const,
+                      searchResults[selectedIndex].name,
+                      {
+                        shouldTouch: true,
+                      },
+                    );
+                    setValue(
+                      'dataBlockParentId' as const,
+                      searchResults[selectedIndex].dataBlockParentId,
+                      { shouldTouch: true },
+                    );
+                    trigger('dataBlockParentId');
+                    setSelectedEntry(searchResults[selectedIndex]);
+                  }}
+                />
+
+                {getValues('dataBlockParentId') && (
+                  <RHFFormFieldTextInput<FormValues>
+                    formGroupClass="govuk-!-margin-top-5 govuk-!-margin-bottom-2"
+                    id="text"
+                    label="Link text"
+                    name="text"
+                  />
+                )}
+
+                <ButtonGroup className="govuk-!-margin-top-5">
+                  <Button
+                    type="submit"
+                    onClick={event => {
+                      // Have to prevent default here as the form is nested,
```

Comment:
Unfortunately it does need to be in the form (which is several layers up from this component in `EditableContentForm`) so that it has access to the CKEditor methods. When it renders it's not actually within the form in the DOM, it's just an annoyance of React that it still triggers the form submission.

Reply:
Gotcha thanks.